### PR TITLE
Port imgbuild tests to new execution context

### LINKED ISSUE #3832: Port e2e imgbuild tests
Port imgbuild tests https://github.com/sylabs/singularity/blob/master/e2e/imgbuild/imgbuild.go to use singularity execution context (https://github.com/sylabs/singularity/blob/master/e2e/internal/e2e/singularitycmd.go)

### PR DESCRIPTION
**Pull request by ikaneshiro**
Signed-off-by: Ian Kaneshiro <iankane@umich.edu>

**This fixes or addresses the following GitHub issues:**

- Fixes #3832 

**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](https://github.com/sylabs/singularity/blob/master/CHANGELOG.md) if necessary according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](https://github.com/sylabs/singularity/blob/master/CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](https://github.com/sylabs/singularity/blob/master/CONTRIBUTORS.md)

Attn: @singularity-maintainers

### CODE REVIEW

File: e2e/imgbuild/imgbuild.go [hunk 1]
```diff
@@ -44,26 +44,34 @@ func (c *imgBuildTests) buildFrom(t *testing.T) {
...
-			if tt.dependency != "" {
-				if _, err := exec.LookPath(tt.dependency); err != nil {
-					t.Skipf("%v not found in path", tt.dependency)
-				}
-			}
-
-			opts := e2e.BuildOpts{
-				Sandbox: tt.sandbox,
+		if tt.dependency != "" {
+			if _, err := exec.LookPath(tt.dependency); err != nil {
```
Comment (cclerget) [inline e2e/imgbuild/imgbuild.go]: To not skip next tests, the check need to be wrapped around line 67 in :
```
e2e.PreRun(func(t *testing.T) {
     if tt.dependency != "" {
         ...
     }
})
```

File: e2e/imgbuild/imgbuild.go [hunk 2]
```diff
@@ -119,39 +127,52 @@ func (c *imgBuildTests) buildLocalImage(t *testing.T) {
...
-	})(t)
+	imagePath := path.Join(c.env.TestDir, "container")
+	e2e.RunSingularity(
+		t,
+		"bad path",
+		e2e.WithPrivileges(true),
+		e2e.WithCommand("build"),
+		e2e.WithArgs(imagePath, "/some/dumb/path"),
+		e2e.PostRun(func(t *testing.T) {
+			os.Remove(imagePath)
```
Comment (cclerget) [inline e2e/imgbuild/imgbuild.go]: is it required as the build fail ?

**Review by cclerget — CHANGES_REQUESTED** [review]

File: e2e/imgbuild/imgbuild.go (hunk 2 above)
Comment (ikaneshiro) [reply e2e/imgbuild/imgbuild.go]: Not required

**Review by cclerget — APPROVED** [review]
LGTM

**Pull request merged**